client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala

```diff
@@ -235,29 +235,32 @@ class LifecycleManager(val appUniqueId: String, val conf: CelebornConf) extends
 
     case pb: PbRevive =>
       val shuffleId = pb.getShuffleId
-      val mapId = pb.getMapId
-      val attemptId = pb.getAttemptId
-      val partitionId = pb.getPartitionId
-      val epoch = pb.getEpoch
-      val oldPartition =
-        if (pb.hasOldPartition) {
-          PbSerDeUtils.fromPbPartitionLocation(pb.getOldPartition)
+      val mapIds = pb.getMapIdList
+      val partitionInfos = pb.getPartitionInfoList
+
+      val partitionIds = new util.ArrayList[Integer](partitionInfos.size)
+      val epochs = new util.ArrayList[Integer](partitionInfos.size())
+      val oldPartitions = new util.ArrayList[PartitionLocation](partitionInfos.size());
+      val causes = new util.ArrayList[StatusCode](partitionInfos.size());
+      (0 until partitionInfos.size()).foreach { idx =>
+        val info = partitionInfos.get(idx)
+        partitionIds.add(info.getPartitionId)
+        epochs.add(info.getEpoch)
+        if (info.hasPartition) {
+          oldPartitions.add(PbSerDeUtils.fromPbPartitionLocation(info.getPartition))
         } else {
-          null
+          oldPartitions.add(null)
```

Comment:
this null propagation looks dangerous

Reply:
This has been discussed offline, the main reason is to keep consistent with the following code
```

    // get location
    // If rerun or speculation task running after LifecycleManager call stageEnd,
    // register shuffle will return an empty location map, client need revive for a new location.
    if (!map.containsKey(partitionId)) {
      if (!revive(
          shuffleId,
          mapId,
          attemptId,
          partitionId,
          -1,
          null,
          StatusCode.PUSH_DATA_FAIL_NON_CRITICAL_CAUSE)) {
        throw new CelebornIOException(
            String.format("Revive for shuffle %s partition %d failed.", shuffleId, partitionId));
      }
    }
```
oldPartition can be null